### PR TITLE
Better Chainstart Logging

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

UX Feature

**What does this PR do? Why is it needed?**

- [x] In the event chainstart has not been reached we log the number of validators and
time left till we reach the threshold.
- [x] Remove deposit root field as it is technical debt carried over from 
our old powchain implementation
- [x] Add unit test.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/powchain/log_processing.go [hunk 1]
```diff
@@ -455,17 +462,26 @@ func (s *Service) checkHeaderRange(start uint64, end uint64,
 	return nil
 }
 
-func (s *Service) checkForChainstart(blockHash [32]byte, blockNumber *big.Int, blockTime uint64) {
+func (s *Service) currentCountAndGenesisTime(blockTime uint64) (uint64, uint64) {
```
Comment (rkapka) [inline beacon-chain/powchain/log_processing.go:465]:
```suggestion
func (s *Service) currentValidatorCountAndGenesisTime(blockTime uint64) (uint64, uint64) {
```

File: beacon-chain/powchain/log_processing.go [hunk 2]
```diff
@@ -455,17 +462,26 @@ func (s *Service) checkHeaderRange(start uint64, end uint64,
...
 	valCount, err := helpers.ActiveValidatorCount(s.preGenesisState, 0)
 	if err != nil {
 		log.WithError(err).Error("Could not determine active validator count from pre genesis state")
+		return 0, 0
+	}
+	return valCount, s.createGenesisTime(blockTime)
+}
+
+func (s *Service) checkForChainstart(blockHash [32]byte, blockNumber *big.Int, blockTime uint64) {
+	valCount, genesisTime := s.currentCountAndGenesisTime(blockTime)
```
Comment (rkapka) [inline beacon-chain/powchain/log_processing.go:478]:
```suggestion
    valCount, genesisTime := s.currentValidatorCountAndGenesisTime(blockTime)
```

File: beacon-chain/powchain/service.go [hunk 3]
```diff
@@ -704,6 +692,37 @@ func (s *Service) run(done <-chan struct{}) {
...
+func (s *Service) logTillChainStart() {
+	if s.chainStartData.Chainstarted {
+		return
+	}
+	_, blockTime, err := s.retrieveBlockHashAndTime(s.ctx, big.NewInt(int64(s.latestEth1Data.LastRequestedBlock)))
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	valCount, genesisTime := s.currentCountAndGenesisTime(blockTime)
```
Comment (rkapka) [inline beacon-chain/powchain/service.go:715]:
```suggestion
    valCount, genesisTime := s.currentValidatorCountAndGenesisTime(blockTime)
```

File: beacon-chain/powchain/service.go [hunk 4]
```diff
@@ -704,6 +692,37 @@ func (s *Service) run(done <-chan struct{}) {
...
+	if valCount < params.BeaconConfig().MinGenesisActiveValidatorCount {
+		valNeeded = params.BeaconConfig().MinGenesisActiveValidatorCount - valCount
+	}
+	secondsLeft := uint64(0)
+	if genesisTime < params.BeaconConfig().MinGenesisTime {
+		secondsLeft = params.BeaconConfig().MinGenesisTime - genesisTime
+	}
+	log.WithFields(logrus.Fields{
+		"Extra validators needed":      valNeeded,
+		"Minutes till minimum genesis": uint64((time.Duration(secondsLeft) * time.Second).Minutes()),
```
Comment (rkapka) [inline beacon-chain/powchain/service.go:726]: How about more flexibility? Show days, hours or minutes depending on how much time is left. "24234 minutes" does not tell me much.

**Comment by rkapka** [comment]
Submitting suggestions from GoLand is annoying because of indenting...

File: beacon-chain/powchain/log_processing.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/powchain/log_processing.go:465]: this seems excessively verbose, especially given that this method is only used in this package.

File: beacon-chain/powchain/service.go (hunk 4 above)
Comment (nisdas) [reply beacon-chain/powchain/service.go:726]: good point, I can add that in

File: beacon-chain/powchain/log_processing.go (hunk 1 above)
Comment (rkapka) [reply beacon-chain/powchain/log_processing.go:465]: No problem. It's just that I saw the method name and needed to read the body to understand what does `Count` refer to.

**Review by rauljordan — APPROVED** [review]

File: beacon-chain/powchain/service.go (hunk 4 above)
Comment (nisdas) [reply beacon-chain/powchain/service.go]: Added it in

File: beacon-chain/powchain/log_processing.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/powchain/log_processing.go]: added in a comment

**Review by rauljordan — APPROVED** [review]

**Pull request merged**